lib/ansible/plugins/lookup/sequence.py

```diff
@@ -15,6 +15,7 @@
       - 'Arguments can be specified as key=value pair strings or as a shortcut form of the arguments string is also accepted: [start-]end[/stride][:format].'
       - 'Numerical values can be specified in decimal, hexadecimal (0x3f8) or octal (0600).'
       - Starting at version 1.9.2, negative strides are allowed.
+      - Generated items are strings. Use Jinja2 filters to convert items to preferred type, e.g. ``{{ 1 + item|int }}``.
```

Comment:
This is a bit incorrect, like for all lookups, generated items might be a specific type (in this case, `ints`) but by default JInja2 converts everything into strings so you either enabled 'native types' or use casting on consumption.

This should be a general warning in the main lookup plugin page

Reply:
https://github.com/ansible/ansible/blob/222adde00e0ac2e73addd23be77c5afbd7ae6291/lib/ansible/plugins/lookup/sequence.py#L228

Reply:
No, this is sequence lookup specific. All items are formatted (even if you don't specify the format) and returned as strings. Jinja2 comes into play later and fails if you try to do unsupported operation on strings, native types won't help with this.

Reply:
ah, true, i forgot the `format` option forces always to stringify

Reply:
I would also point out that the `range` filter is a better alternative